Posts/2021/10/tfl-secrets.md

```diff
@@ -0,0 +1,228 @@
+You could, in theory, change the secret store and not need to change your scripts that use the secrets.
+
+## Installing the Modules
+
+```powershell-console
+PS> # 1. Discover the modules
+PS> $Names =  'Microsoft.PowerShell.SecretManagement',
+              'Microsoft.PowerShell.SecretStore'
+PS> Find-Module -Name $Names |
+      Format-Table -Wrap -AutoSize
+
+Version Name                                  Repository Description
+------- ----                                  ---------- -----------
+1.1.0   Microsoft.PowerShell.SecretManagement PSGallery  This module provides a convenient way for a user
+                                                         to store and retrieve secrets. The secrets are
+                                                         stored in registered extension vaults. An 
+                                                         extension vault can store secrets locally or remotely.
+                                                         SecretManagement coordinates access to the secrets
+                                                         through the registered vaults.
+                                                         Go to GitHub for more information about the module
+                                                         and to submit issues:https:////github.com//powershell//SecretManagement
+
+1.0.4   Microsoft.PowerShell.SecretStore      PSGallery  This PowerShell module is an extension vault for the
+                                                         PowerShell SecretManagement module.
+                                                         As an extension vault, this module stores secrets to the local
+                                                         machine based on the current user account context. 
+                                                         The secrets are encrypted on file using .NETCrypto APIs. 
+                                                         A password is required in the default configuration. 
+                                                         The configuration can be changed with the provided cmdlets.
+                                                         Go to GitHub for more information about this module 
+                                                         and to submit issues: https:////github.com//powershell//SecretStore
+
+PS> # 2. Install both modules
+PS> Install-Module -Name $Names -Force -AllowClobber
+```
+When you install the module using `Install-Module` you see no output (unless you use the `-Verbose` switch).
+You can always use `Get-Module` to check that you have installed these new (to you) modules.
+
+## Discovering the commands available to you
+Once you have thess two modules installed, you can discover the commands in each module:
+
+```powershell-console
+
+PS> # 3. Examine them
+PS> Get-Module -Name Microsoft*.Secret* -List |
+       Format-Table -Property ModuleType*, Ver*, Name, Ps*, Exportedc*
```

Comment:
Not sure why you are using so many wildcards here. `PS*` doesn't match anything - the property is **CompatiblePSEditions**. `Exportedc*` matches **ExportedCmdlets** and **ExportedCommands** but only one is displayed here. Also, you might consider using `Format-List` here instead.
```suggestion
PS> Get-Module -Name Microsoft*.Secret* -ListAvailable |
       Format-Table -Property ModuleType, Version, Name, ExportedCmdlets
```